controllers/toolchaincluster/toolchaincluster_controller_test.go

```diff
@@ -119,6 +124,53 @@ func TestClusterControllerChecks(t *testing.T) {
 		require.NoError(t, err)
 		assertClusterStatus(t, cl, "stable", offline())
 	})
+
+	t.Run("migrates connection settings to kubeconfig in secret", func(t *testing.T) {
+		// given
+		tc, secret := newToolchainCluster("tc", tcNs, "http://cluster.com", toolchainv1alpha1.ToolchainClusterStatus{})
+		cl := test.NewFakeClient(t, tc, secret)
+		reset := setupCachedClusters(t, cl, tc)
+		defer reset()
+
+		controller, req := prepareReconcile(tc, cl, requeAfter)
+		expectedKubeConfig := composeKubeConfigFromData([]byte("mycooltoken"), "http://cluster.com", "test-namespace", true)
+
+		// when
+		_, err := controller.Reconcile(context.TODO(), req)
+		secretAfterReconcile := &corev1.Secret{}
+		require.NoError(t, cl.Get(context.TODO(), client.ObjectKeyFromObject(secret), secretAfterReconcile))
+		actualKubeConfig, loadErr := clientcmd.Load(secretAfterReconcile.Data["kubeconfig"])
+
+		// then
+		require.NoError(t, err)
+		require.NoError(t, loadErr)
+		assert.Contains(t, secretAfterReconcile.Data, "kubeconfig")
+
+		// we need to use this more complex equals, because we don't initialize the Extension fields (i.e. they're nil)
+		// while they're initialized and empty after deserialization, which causes the "normal" deep equals to fail.
+		result := cmp.DeepEqual(expectedKubeConfig, *actualKubeConfig,
+			cmpopts.IgnoreFields(clientcmdapi.Config{}, "Extensions"),
+			cmpopts.IgnoreFields(clientcmdapi.Preferences{}, "Extensions"),
+			cmpopts.IgnoreFields(clientcmdapi.Cluster{}, "Extensions"),
+			cmpopts.IgnoreFields(clientcmdapi.AuthInfo{}, "Extensions"),
+			cmpopts.IgnoreFields(clientcmdapi.Context{}, "Extensions"),
+		)()
+
+		assert.True(t, result.Success())
+	})
+}
+
+func TestComposeKubeConfig(t *testing.T) {
```

Comment:
should we also check for the keys in kubeconfig: 
- `ctx` ( which is the default context I see )
- `cluster` ( which is the default cluster )
- `auth` ( which holds the authentication info ) 

or those are not going to be referenced anywhere ?

Reply:
These are all tested by the means of the `CurrentContext`. This is set to "ctx" and the context then references the cluster and auth by name and as such binds everything together.